### ISSUE TITLE
Aggregations by capture group are case-sensitive even when queries are not

### ISSUE THREAD

**Issue by felixfbecker** [sourcegraph/sourcegraph#41018]
When the query is case-insensitive, the aggregation by capture group is not:

<img width="251" alt="image" src="https://user-images.githubusercontent.com/10532611/187288176-1718fb91-0a14-4de3-9db0-7660c35e1868.png">

This leads to the unexpected case that when drilling down into a bar, it matches more results than the count of the bar, because it will also match other cases.

The expected behavior would be (I think) that the aggregation is also case-insensitive, i.e. collects all matches into a single bar `test`.

/cc @joelkw @felixfbecker @vovakulikov @unclejustin

**Comment by chwarwick** [comment]
This is an interesting case, I'm not sure what I would expect here as a user.  I think it would look strange if we were to change the casing on all the results.

**Comment by rvantonder** [comment]
Probably related to https://github.com/sourcegraph/sourcegraph/issues/37972 (renamed for accuracy)

**Comment by chwarwick** [comment]
> Probably related to #37972 (renamed for accuracy)

@rvantonder This is unrelated as we directly using search results for these aggregations.  In this case we got the casing right when matching but didn't carry it forward to when we bin the results.

**Comment by leonore** [comment]
I'm not sure that as a user I would expect the capture group matches to be case insensitive. 

I think I would expect
1. capture group aggregations to be shown case sensitive (i.e. test, TEST, test, which is currently the case)
2. drilldowns to be case sensitive (i.e. when clicking into test don't match TEST, which is **not currently the case**)

example query to drill down into: `file:test ([a-zA-z]+)Schema file:batches`

however I think for the sake of this feature we should go with what's easiest to implement. but at the moment there is a mismatch between 1 and 2

**Comment by leonore** [comment]
I've actually found `case` searches are not respected because the parameter is not passed in to the query. created https://github.com/sourcegraph/sourcegraph/issues/41120

**Comment by Joelkw** [comment]
notes from sync: low confidence in what this output should be, lowest priority.

**Comment by leonore** [comment]
behaviour isn’t changed now that case sensitive queries are respected in aggregations, my [comment above](https://github.com/sourcegraph/sourcegraph/issues/41018#issuecomment-1232828418) still stands. 

however I'd be more inclined to 1. display different case matches in aggregations (because it seems to add some value to the search to me) and 2. make capture group drilldowns case sensitive, because it feels weird to click into `new` and receive results for `New` as Felix initially mentioned:

> This leads to the unexpected case that when drilling down into a bar, it matches more results than the count of the bar, because it will also match other cases.

however I'd love to get another backend-dev opinion on this. it would require some refactoring on the backend to write `(?-i:)`afaics and I wouldn't want to break anything else

**Comment by coury-clark** [comment]
As a parallel thought, consider that we've already taken the position that these aggregations are not coupled to the original query in the case of `count` limits. Given that, I agree with the proposal from @leonore to display case sensitive aggregations regardless of search mode (possibly limiting this to capture group aggregations).

1. It's following the same logic as the `count:all` expansion, that we want to add more value to the aggregation than the pure search result
2. It still allows users to be more specific by enabling `case:yes`

In other words, the behavior of aggregations right now is the most value providing and complete, with the problem that we need to produce drilldown queries with `case:yes`

**Comment by coury-clark** [comment]
[Here is a PR if thats the direction we take](https://github.com/sourcegraph/sourcegraph/pull/41392)

**Comment by coury-clark** [comment]
@leonore I'm going to steal this issue then and also use this chance to add some tests that I wanted to circle back to here. Will put up a PR later
